code/modules/mob/living/living.dm

```diff
@@ -952,4 +952,15 @@
 
 //used in datum/reagents/reaction() proc
 /mob/living/proc/get_permeability_protection()
-	return 0
\ No newline at end of file
+	return 0
+
+/mob/living/proc/harvest(mob/living/user)
+	if(qdeleted(src))
+		return
+	if(butcher_results)
+		for(var/path in butcher_results)
+			for(var/i = 1, i <= butcher_results[path], i++)
+				new path(loc)
+			butcher_results.Remove(path) //In case you want to have things like simple_animals drop their butcher results on gib, so it won't double up below.
```

Comment:
@Fox-McCloud this is what keeps it from doubling after harvesting..it can gib and drop its stuff...or you can butcher it, when you butcher it it removes the butcher object from the list so its not used again